pkg/proto/nbcontract/v1/identityconfig.proto

```diff
@@ -12,6 +12,6 @@ message IdentityConfig {
 enum IdentityType {
   IDENTITY_TYPE_UNSPECIFIED = 0;
   IDENTITY_TYPE_SERVICE_PRINCIPAL = 1;
-  IDENTITY_TYPE_SYSTEM_IDENTITY = 2;
-  IDENTITY_TYPE_USER_IDENTITY = 3;
+  IDENTITY_TYPE_USER_IDENTITY = 2;
```

Comment:
Is it best practice to prefix these with IDENTITY_TYPE_? Or would this be clear / already namespaces because of "IdentityType" enum?

Reply:
hmm. This is good point. In Karpenter now we are using it as
_identityConfig.IdentityType = nbcontractv1.IdentityType_IDENTITY_TYPE_USER_IDENTITY_
So I think you are right. We can remove the prefix. I will review other enums too.

Reply:
Oh I know why we needed to keep the prefix.
For example, if I put

`enum NetworkPluginType {
UNSPECIFIED = 0;
NONE = 1;
AZURE = 2;
KUBENET = 3;
}

enum NetworkPolicyType {
UNSPECIFIED = 0;
NONE = 1;
AZURE = 2;
CALICO = 3;
}`

It will complain this.
`symbol "nbcontract.v1.AZURE" already defined at pkg/proto/nbcontract/v1/networkconfig.proto:32:3; protobuf uses C++ scoping rules for enum values, so they exist in the scope enclosing the enum`

Reply:
protobuf spec [for enums](https://protobuf.dev/reference/go/go-generated/#enum) says constants generated for enums _within a message_ are prefixed with the message name automatically. Worth a try?

Reply:
I tried this.

```
message NetworkConfig {
  NetworkPluginType network_plugin = 1;
  NetworkPolicyType network_policy = 2;
  NetworkModeType network_mode = 3;
  
  enum NetworkPluginType {
    UNSPECIFIED = 0;
    NONE = 1;
    AZURE = 2;
    KUBENET = 3;
  }

  enum NetworkPolicyType {
    UNSPECIFIED = 0;
    NONE = 1;
    AZURE = 2;
    CALICO = 3;
  }
}
```

It still complains 
```
pkg/proto/nbcontract/v1/networkconfig.proto:24:5:symbol "nbcontract.v1.NetworkConfig.UNSPECIFIED" already defined at pkg/proto/nbcontract/v1/networkconfig.proto:17:5; protobuf uses C++ scoping rules for enum values, so they exist in the scope enclosing the enum
pkg/proto/nbcontract/v1/networkconfig.proto:25:5:symbol "nbcontract.v1.NetworkConfig.NONE" already defined at pkg/proto/nbcontract/v1/networkconfig.proto:18:5; protobuf uses C++ scoping rules for enum values, so they exist in the scope enclosing the enum
pkg/proto/nbcontract/v1/networkconfig.proto:26:5:symbol "nbcontract.v1.NetworkConfig.AZURE" already defined at pkg/proto/nbcontract/v1/networkconfig.proto:19:5; protobuf uses C++ scoping rules for enum values, so they exist in the scope enclosing the enum
make: *** [Makefile:90: compile-proto-files] Error 100
```